task6/Working with posts.js

```diff
@@ -19,86 +19,91 @@ class post {
+new Post("24", "Ava.jpg", "incredible4",  new Date('2020-12-07T10:14:22'), "Hello111"),
+new Post("25", "Ava.jpg", "incredible4",  new Date('2020-12-15T02:38:40'), "Hello481"),
+new Post("26", "Ava.jpg", "incredible4",  new Date('2020-01-05T17:41:38'), "Hello211"),
+new Post("27", "Ava.jpg", "incredible4",  new Date('2020-06-07T13:53:03'), "Hello154", ["#hello", "#hai"]),
+new Post("28", "Ava.jpg", "incredible4",  new Date('2020-01-23T12:15:11'), "Hello564"),
+new Post("29", "Ava.jpg", "incredible4",  new Date('2020-07-05T20:30:55'), "Hello931"),
+new Post("30", "Ava.jpg", "incredible4",  new Date('2020-04-25T00:08:16'), "Hello990"),
+new Post("31", "Ava.jpg", "incredible5",  new Date('2020-07-10T09:20:03'), "Hello633"),
+new Post("32", "Ava.jpg", "incredible5",  new Date('2020-09-27T18:55:54'), "Hello468"),
+new Post("33", "Ava.jpg", "incredible5",  new Date('2020-12-27T12:39:01'), "Hello530", ["#hai"]),
+new Post("34", "Ava.jpg", "incredible5",  new Date('2020-12-20T18:54:23'), "Hello927", ["#ni hao"]),
+new Post("35", "Ava.jpg", "incredible5",  new Date('2020-07-19T11:25:13'), "Hello549"),
+new Post("36", "Ava.jpg", "incredible5",  new Date('2020-10-26T40:03:24'), "Hello616"),
+new Post("37", "Ava.jpg", "incredible5",  new Date('2020-12-23T17:20:30'), "Hello682"),
+new Post("38", "Ava.jpg", "incredible5",  new Date('2020-07-28T00:46:26'), "Hello208"),
+new Post("39", "Ava.jpg", "incredible5",  new Date('2020-10-22T02:59:26'), "Hello940"),
+new Post("40", "Ava.jpg", "incredible5",  new Date('2020-03-15T21:40:13'), "Hello999")
 ];
 
 
-class editInfo{
+class EditInfo{
 	constructor(description, hashTags = [], photoLink="") {
 		this.description = description;
 		this.hashTags = hashTags;
 		this.photoLink = photoLink;
 	}
 }
 
-class filterParams {
-	constructor (author, date, hashTags=[]) {
+class DateInterval {
+	constructor (left, right) {
+		this.left = left;
+		this.right = right;
+	}
+}
+
+class FilterParams {
+	constructor (author, dateInterval, hashTags=[]) {
 		this.author = author;
-		this.date = date;
+		this.dateInterval = dateInterval;
 		this.hashTags = hashTags;
 	}
 }
 
-function inOneDay(date1, date2) {
-	return date1.getFullYear() == date2.getFullYear()
-		 && date1.getMonth() == date2.getMonth()
-		 && date1.getDate() == date2.getDate();
+function liesInInterval(date, dateInterval) {
+	return dateInterval.left <= date && date <= dateInterval.right;
 }
 
 function hasAnyTag(hashTagsOfPost, hashTags) {
-	return hashTagsOfPost.some(hashTagOfPost=>(hashTags.some(hashTag=>hashTag==hashTagOfPost)));
+	return hashTagsOfPost.some(hashTagOfPost=>(hashTags.includes(hashTagOfPost)));
 }
 
 function getPostsByFilter(filterConfig) {
 	return posts.filter(element=>
-		(element.author == filterConfig.author) &&
-		((filterConfig.date == null) || inOneDay(element.createdAt, filterConfig.date)) &&
+		(filterConfig.author == "" || element.author == filterConfig.author) &&
```

Comment:
Replace `filterConfig.author == ""` with `!filterconfig.autor` since it can be undefined or null